plugins/modules/vmware_vmkernel.py

```diff
@@ -792,7 +787,7 @@ def find_dvspg_by_key(self, dv_switch, portgroup_key):
 
         return None
 
-    def set_vsan_service_type(self):
+    def set_vsan_service_type(self, enable_vsan=True):
```

Comment:
It looks like you're changing the default behavior here. Both the documentation and the `argument_spec` state that the default for `enable_vsan` is `false`.

This would be a breaking change and we need some time to announce it. We can't do it just like that from one minor version to another. See #739 for a similar issue.

This would also mean that `enable_vsan` would work differently from all the other `enable_something` parameters which looks like a bad idea to me ([POLA](https://en.wikipedia.org/wiki/Principle_of_least_astonishment)).

Reply:
No, this specific method "set_vsan_service_type" is called only in two cases :
- Line 930 : where there is an explicit check for self.enable_vsan (the actual parameter for the module ; sorry if this is confusing, but the method parameter used at this point in the code is not the same as the module parameter the user will configure
- Line 754 : if the vSAN flag changed for this vmknic (in which case the module parameter is passed)

I assumed that the initial method would only be called when it was already clear that the user intended to enable vSAN for a specific vmknic, and added the extra parameter and its default value with this in mind.

Regarding your other question, I have since tried the latest version of Ansible (installed manually via pip; which is a bit of a problem to run code in production in an air-gapped environment where we are stuck with RedHat packages), but no, this was not fixed.

Regarding your worries, maybe I need to change the parameter name for this method from "enable_vsan" to simply "enable_for_vmknic", but I think the logic itself is sound and my understanding is that the specification itself does not change. (i.e: if you don't run a task with "enable_vsan", it won't get enabled)

I could also flip the logic on its head, set my method parameter to False as a default, and be explicit in passing self.enable_vsan (even though we know it to be True by this point, since we just checked it)

Reply:
I decided to go with the shortest fix, which should make the method safe to re-use elsewhere just in case : 
```diff
diff --git a/plugins/modules/vmware_vmkernel.py b/plugins/modules/vmware_vmkernel.py
index 2d97d1f..67c79f2 100644
--- a/plugins/modules/vmware_vmkernel.py
+++ b/plugins/modules/vmware_vmkernel.py
@@ -787,7 +787,7 @@ class PyVmomiHelper(PyVmomi):
 
         return None
 
-    def set_vsan_service_type(self, enable_vsan=True):
+    def set_vsan_service_type(self, enable_vsan=False):
         """
         Set VSAN service type
         Returns: result of UpdateVsan_Task
@@ -927,7 +927,7 @@ class PyVmomiHelper(PyVmomi):
 
             # VSAN
             if self.enable_vsan:
-                results['vsan'] = self.set_vsan_service_type()
+                results['vsan'] = self.set_vsan_service_type(self.enable_vsan)
 
             # Other service type
             host_vnic_manager = self.esxi_host_obj.configManager.virtualNicManager
```

Reply:
Thanks @darksoul42, I've resolved this conversation for now.

However, I haven't worked on this module yet (at least afair) so I don't feel confident enough to approve your change. I hope I'll find the time to dig deeper into this during the next days.

> Regarding your other question, I have since tried the latest version of Ansible (installed manually via pip; which is a bit of a problem to run code in production in an air-gapped environment where we are stuck with RedHat packages), but no, this was not fixed.

Thank you very much for testing this! So it really _does_ look like there's (still) a bug in the code that hasn't been fixed yet.